### ISSUE TITLE
Mutating features in localhost mode does not work to update a treatment.

### ISSUE THREAD

**Issue by dreid** [splitio/javascript-browser-client#119]
The documentation of localhost mode here: https://help.split.io/hc/en-us/articles/360058730852-Browser-SDK#localhost-mode

says:

> You can then change the treatment as necessary for your testing. To update a treatment or a config, or add or remove feature flags from the mock cache, update the features object you've provided.

This implies that the browser SDK is retaining the reference to the passed features object however it appears that features actually gets copied when it passes through a deep copy merge implementation used by `settingsValidation` here: https://github.com/splitio/javascript-commons/blob/e5eca00b8df9bbda1a1fb7925a7e1663f8ac0552/src/utils/settingsValidation/index.ts#L111

**Comment by EmilianoSanchez** [comment]
Hi @dreid , 

Thank you for sharing. You are right, the SDK does a deep copy of the `config` object, including its `features` property. So you cannot do as follows as the docs suggest:

```javascript
const config = { ..., features: { 'test_flag': 'on' } };
const factory = SplitFactory(config);

config.features['test_flag'] = 'off'; // It will NOT emit SDK_UPDATE
```

At the moment, if you want to trigger an SDK_UPDATE when using localhost mode, you can update the internal copy instead, as follows:

```
const config = { ..., features: { 'test_flag': 'on' } };
const factory = SplitFactory(config);

factory.settings.features = { 'test_flag': 'off' }; // It will emit SDK_UPDATE considering `scheduler.offlineRefreshRate` option, which is 15 seconds by default
```

However we will update the SDK behaviour to support both cases and match the documentation. In the meantime, you can use the workaround above. 

I will keep you updated and will close this issue as soon as we release a new version with the fix.

Thanks again for your feedback.

**Comment by EmilianoSanchez** [comment]
Hi @dreid, 

We have released v0.14.1 of the JS Browser SDK with some updates on localhost mode to match what the docs say. Take a look at the following example to see how you can update the `features` object map since v0.14.1:

```javascript
import { SplitFactory } from '@splitsoftware/splitio-browserjs/full'
const config = {
  core: {
    authorizationKey: 'localhost',
    key: 'anonymous',
  },
  features: {
    feature1: 'on'
  }
};

const factory = SplitFactory(config);

// The SDK keeps a reference to the `features` object map of the provided `config`, so you can mutate the object as follows to emit SDK_UPDATE events:
config.features['feature1'] = 'off'; // update feature1
config.features['feature2'] = 'off'; // add feature2
delete config.features['feature1']; // delete feature1

// In case you need to update the whole mock object, you can replace the factory internal reference:
factory.settings.features = { 'feature1': 'on' };

// But you cannot do it on the passed configuration, as the SDK will not reference the new object:
config.features = { 'feature1': 'off' }; // Will not emit SDK_UPDATE
```

We will eventually add some similar code snippet examples in our docs.

I am closing the issue, but feel free to reach out if you have any other questions.
